### PR TITLE
Close Function and event listener added

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
#72 [Bug] add close method to database and a process exit event handler for calling it

files impacted:

- app.ts
- mongodb.ts
- database.ts
- inversify.config.ts

Attached result:
Screenshot from 2020-03-30 22-02-19

### CODE REVIEW

File: Notifier-service/src/app.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ import { Logger } from './utility/log';
 import { Route } from './interfaces/route';
 import { Listener } from './interfaces/listener';
 import { AppContainer } from './inversify.config';
+import { Database } from './interfaces/database';
+import { async } from 'rxjs/internal/scheduler/async';
```
Comment (rajatkb) [inline Notifier-service/src/app.ts:9]: * Not needed

File: Notifier-service/src/database/mongodb.ts [hunk 2]
```diff
@@ -1,28 +1,38 @@
...
       super()
       let databaseName = process.env.MONGO_DB_NAME
       if(databaseName == undefined)
          throw new Error("No database name provided in environment")
       this.dbName = databaseName
+      this.databaseConnection = mongoose.createConnection(`mongodb://${process.env.MONGO_DB_HOST}:${process.env.MONGO_DB_PORT}/${this.dbName}` ,
+      {  useNewUrlParser: true ,
+         useUnifiedTopology: true
+      } )
+      .then((res) => {
```
Comment (rajatkb) [inline Notifier-service/src/database/mongodb.ts:22]: * Add a info log for successfull connection creation.
* add a error log for failed creation in catch.

File: Notifier-service/src/database/mongodb.ts [hunk 3]
```diff
@@ -1,28 +1,38 @@
...
-                            useUnifiedTopology: true
-                         } )
-                         .then((res) => {
-                            return Promise.resolve(res)
-                         })
-                         .catch((err) => {
-                            return Promise.reject(err)
-                         })
+   public async getConnection():Promise<mongoose.Connection>{
+      return await this.databaseConnection;
```
Comment (rajatkb) [inline Notifier-service/src/database/mongodb.ts:30]: * Do not await, simply return.

File: Notifier-service/src/database/mongodb.ts [hunk 4]
```diff
@@ -1,28 +1,38 @@
...
    };
+   public async close():Promise<void>{
+      await this.databaseConnection.then(database => {
+         this.logger.info(">>>>>>>>>>>>>>>>>>>>>>>>>>>> Connection closed!! <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<");
```
Comment (rajatkb) [inline Notifier-service/src/database/mongodb.ts:34]: * You can simply write MongoDatabase Connection close.

**Comment by Kajol-Kumari** [comment]
@rajatkb Done! All the changes

**Pull request merged**